addons/base/views.py

```diff
@@ -345,18 +346,11 @@ def get_auth(auth, **kwargs):
 ])
 
 
-# TODO: Use this to mark file versions as seen when
-# MFR callback endpoint is implemented
-def mark_file_version_as_seen(user, path, version):
+def mark_file_version_as_seen(user, file_to_update, version):
     """
     Mark a file version as seen by the given user.
-    If no version is included, default to the most recent version.
     """
-    file_to_update = OsfStorageFile.objects.get(_id=path)
-    if version:
-        file_version = file_to_update.versions.get(identifier=version)
-    else:
-        file_version = file_to_update.versions.order_by('-created').first()
+    file_version = file_to_update.versions.get(identifier=version)
```

Comment:
It feels a bit strange that to query for the relevant file version here when it's already being computed in `get_auth` (~L300). Why not compute it once and use it for both `mark_file_version_as_seen` and `update_analytics`?